clients/ctl/admin-ui/src/features/dataset/helpers.ts

```diff
@@ -58,6 +61,54 @@ export const getUpdatedDatasetFromField = (
   );
 };
 
+/**
+ * Returns a new dataset object with the top-scoring classification results filling in data
+ * categories that were left blank on the dataset. Uses immer to efficiently modify a draft object.
+ */
```

Comment:
oh that's nice, I guess this would also work well for the other functions in this file too. those were really annoying to write...